ml/jml/probabilizer.h

```diff
@@ -332,6 +336,31 @@ operator << (DB::Store_Writer & store, const GLZ_Probabilizer & prob);
 DB::Store_Reader &
 operator >> (DB::Store_Reader & store, GLZ_Probabilizer & prob);
 
+/*****************************************************************************/
+/* PROBABILIZER                                                          */
+/*****************************************************************************/
+
+struct ProbabilizerModel
+{
+    ProbabilizerModel();
+    ProbabilizerModel(std::shared_ptr<spdlog::logger> logger);
+
+    //Score, label, weight
+    void train(const std::vector<std::tuple<float, float, float> >& fvs,
+               ML::Link_Function link);
+
+    void serialize(DB::Store_Writer & store) const;
+    void reconstitute(DB::Store_Reader & store);
+
+    std::shared_ptr<spdlog::logger> logger;
```

Comment:
Ugh, no thanks.  We need to get out of the habit of putting shared pointers to loggers everywhere.  If we want to log, then we should get the logger at the point of logging, or pass it in, or make it thread-local, or something.